worker/secretmigrationworker/worker.go

```diff
@@ -0,0 +1,202 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package secretmigrationworker
+
+import (
+	"context"
+
+	"github.com/juju/errors"
+	"github.com/juju/worker/v3"
+	"github.com/juju/worker/v3/catacomb"
+
+	coresecrets "github.com/juju/juju/core/secrets"
+	"github.com/juju/juju/core/watcher"
+	jujusecrets "github.com/juju/juju/secrets"
+)
+
+// logger is here to stop the desire of creating a package level logger.
+// Don't do this, instead use the one passed as manifold config.
+type logger interface{}
+
+var _ logger = struct{}{}
+
+// Logger represents the methods used by the worker to log information.
+type Logger interface {
+	Tracef(string, ...interface{})
+	Debugf(string, ...interface{})
+	Warningf(string, ...interface{})
+	Infof(string, ...interface{})
+	Errorf(string, ...interface{})
+	Criticalf(string, ...interface{})
+}
+
+// Facade instances provide a set of API for the worker to deal with secret migration task changes.
+type Facade interface {
+	WatchSecretBackendChanged() (watcher.NotifyWatcher, error)
```

Comment:
The 3 methods here should be on a separate facade. This worker should have 2 facades  - a secrets drain facade with the 3 new methods on it and a JujuAPIClient for processing the changes.